extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultManager.java

```diff
@@ -0,0 +1,60 @@
+package io.quarkus.vault.runtime;
+
+import io.quarkus.vault.runtime.client.OkHttpVaultClient;
+import io.quarkus.vault.runtime.client.VaultClient;
+import io.quarkus.vault.runtime.config.VaultServerRuntimeConfig;
+
+public class VaultManager {
+
+    private static volatile VaultManager instance;
```

Comment:
I think it would be better to manage the lifecycle of these classes with CDI and make them beans.

Reply:
I agree. I wish I could, but I need to use them from the config source. I do not believe this is feasible. do you know if that should work?